src/coreComponents/constitutive/solid/CoupledSolid.hpp

```diff
@@ -0,0 +1,188 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+
+/**
+ * @file CoupledSolid.hpp
+ */
+
+#ifndef GEOSX_CONSTITUTIVE_SOLID_COUPLEDSOLID_HPP_
+#define GEOSX_CONSTITUTIVE_SOLID_COUPLEDSOLID_HPP_
+
+#include "constitutive/ConstitutiveBase.hpp"
+#include "constitutive/solid/SolidBase.hpp"
+
+namespace geosx
+{
+namespace constitutive
+{
+
+template< typename SOLID_TYPE,
+          typename PORO_TYPE >
+class CoupledSolidUpdates
+{
+public:
+  /**
+   * @brief Constructor
+   */
+  CoupledSolidUpdates( SOLID_TYPE * solidModel,
+                       PORO_TYPE * porosityModel ):
+    m_solidUpdate( solidModel->createKernelUpdates() ),
+    m_porosityUpdate( porosityModel->createKernelUpdates() )
+  {}
+
+  /// Deleted default constructor
+  CoupledSolidUpdates() = delete;
+
+  /// Default copy constructor
+  CoupledSolidUpdates( CoupledSolidUpdates const & ) = default;
+
+  /// Default move constructor
+  CoupledSolidUpdates( CoupledSolidUpdates && ) = default;
+
+  /// Deleted copy assignment operator
+  CoupledSolidUpdates & operator=( CoupledSolidUpdates const & ) = delete;
+
+  /// Deleted move assignment operator
+  CoupledSolidUpdates & operator=( CoupledSolidUpdates && ) =  delete;
+
+  GEOSX_HOST_DEVICE
+  GEOSX_FORCE_INLINE
+  typename SOLID_TYPE::KernelWrapper const getSolidModel() const
+  {
+    return m_solidUpdate;
+  }
+
+  GEOSX_HOST_DEVICE
+  GEOSX_FORCE_INLINE
+  typename PORO_TYPE::KernelWrapper const getPorosityModel() const
+  {
+    return m_porosityUpdate;
+  }
```

Comment:
I think the getter will be need in my other PR once I start using the compressible solid in the flow solvers but what do you mean by "not exposing so much" ? You mean just fwd each method that is needed? I will probably need a few more getters in the flow solvers which is why I thought it would be good to give access to the object but I can certainly change it.

Reply:
"_Not exposing too much_" means that you give access to the entire `PORO_TYPE::KernelWrapper` while you only use one single function. It's then more likely that the devs using your code may use the `kernelWrapper` in ways that break your logic. After all, they can 😉

Reply:
I see, makes sense. I will only expose the getters then.